server/models/index.js

```diff
@@ -0,0 +1,42 @@
+const { Sequelize } = require("sequelize");
+const BookModel = require("./book");
+const AuthorModel = require("./author");
+require("dotenv").config({
+  path: "../.env",
+});
+
+const sequelize = new Sequelize(process.env.POSTGRES_URI);
+
+async function createDatabaseIfNotExists() {
+  try {
+    // Check if the database exists
+    const [results, metadata] = await sequelize.query(
+      "SELECT 1 FROM pg_database WHERE datname = 'booksdb'"
+    );
+
+    if (results.length === 0) {
+      // Database doesn't exist, so create it
+      await sequelize.query("CREATE DATABASE booksdb");
+      console.log("Database 'booksdb' created successfully");
+    } else {
+      console.log("Database 'booksdb' already exists");
+    }
+  } catch (error) {
+    console.error("Error checking/creating database:", error);
+  }
+}
+
+createDatabaseIfNotExists();
+
+const Book = BookModel(sequelize);
+const Author = AuthorModel(sequelize);
+
+// Associations
+Author.hasMany(Book, { foreignKey: "author_id" });
+Book.belongsTo(Author, { foreignKey: "author_id", as: "author" });
+
```

Comment:
WOuld would be a got way to co-locate this with the model information and the schema for that model.